web/src/components/filters/__tests__/autocomplete-filter.spec.tsx

```diff
@@ -0,0 +1,78 @@
+import { TextInput, ValidatedOptions } from '@patternfly/react-core';
+import { mount } from 'enzyme';
+import * as React from 'react';
+import { act } from 'react-dom/test-utils';
+import { findFilter } from '../../../utils/filter-definitions';
+import AutocompleteFilter, { AutocompleteFilterProps } from '../autocomplete-filter';
+
+const t = (k: string) => k;
+
+const waitForComponentToPaint = async (): Promise<void> => {
+  await act(async () => {
+    await new Promise(resolve => setTimeout(resolve, 0));
+  });
+};
+
+describe('<AutocompleteFilter />', () => {
+  const props: AutocompleteFilterProps = {
+    filterDefinition: findFilter(t, 'src_name')!,
+    indicator: ValidatedOptions.default,
+    addFilter: jest.fn(),
+    setMessageWithDelay: jest.fn(),
+    setIndicator: jest.fn()
+  };
+  beforeEach(() => {
+    props.addFilter = jest.fn();
+    props.setIndicator = jest.fn();
+  });
+
+  it('should filter valid port by name', async () => {
+    const wrapper = mount(<AutocompleteFilter {...props} filterDefinition={findFilter(t, 'src_port')!} />);
+    const textInput = wrapper.find(TextInput).at(0);
+    const searchButton = wrapper.find('#search-button').at(0);
+
+    expect(textInput).toBeDefined();
+    expect(searchButton).toBeDefined();
+
+    // No initial call
+    expect(props.addFilter).toHaveBeenCalledTimes(0);
+    // Initial setup
+    expect(props.setIndicator).toHaveBeenCalledTimes(2); // <-- WHY ?
```

Comment:
You are using 2 useEffect on the same dependency

Reply:
thanks I'll check that!